src/Access/AccessControl.cpp

```diff
@@ -559,33 +588,43 @@ AccessChangesNotifier & AccessControl::getChangesNotifier()
 }
 
 
-AuthResult AccessControl::authenticate(const Credentials & credentials, const Poco::Net::IPAddress & address) const
+AuthResult AccessControl::authenticate(const Credentials & credentials, const Poco::Net::IPAddress & address, const String & forwarded_address) const
 {
     try
     {
-        return MultipleAccessStorage::authenticate(credentials, address, *external_authenticators, allow_no_password,
-                                                   allow_plaintext_password);
+        const auto auth_result =  MultipleAccessStorage::authenticate(credentials, address, *external_authenticators, allow_no_password,
```

Comment:
I would agree with you, but how do we get the quota without `user_id`?  Do you know a way to get the `user_id` and quota for an authenticated user?

Now AuthResult returns user_id even if authentication is failed, but password check did not pass.

Reply:
> Do you know a way to get the user_id?

`auto user_id = find<User>(credentials.getUserName())`

Reply:
Thanks, If that will also work with users authenticated using LDAP, then we can avoid modification of AuthResult, I will try to implement that way

Reply:
> If that will also work with users authenticated using LDAP

Well, `LDAPAccessStorage::find<User>(credentials.getUserName())` returns nullopt if that user has not logged in before since the server started. So we cannot check the quota this time. But since it's the first time the quota must not be already exceeded. And when the same user logs in again later, `find<User>(credentials.getUserName())` will return a valid `user_id`, so the quota can be checked.

Reply:
It seems for a user who has never logged in successfully using LDAP the quota won't work. Because with only failed attempts `LDAPAccessStorage::authenticate()` will never assign `user_id` to that user and we can't check the quota.

Reply:
I changed the code according to the suggestion to use `find<User>`. And I added a comment for specific LDAP case.